### ISSUE TITLE
Allow to replace fixed string by version

### ISSUE THREAD

**Issue by jobec** [lgiordani/punch#40]
During development, in my documentation, I collect all changes along the way under a section called `next release`. With every release, I rename `next release` to the new version.

Currrently though, that's seems not to be possible, and I need to manually replace that heading. Causing me to frequently forget about it 🙈 

So, something like this would be useful:

```
GLOBALS = {
    'serializer': ["next release", "{{year}}.{{month}}.{{day}}"],
}
```

**Comment by lgiordani** [comment]
Hi! I'm not sure which release is which sorry, can you give me a concrete example with values? I think I already have the solution in the next version I'm developing, but I'm not sure

**Comment by jobec** [comment]
Imagine a CHANGELOG.rst file like this:

```
Changelog
=========

Next Release
------------
**Added**

* Added some new feature
```
What I would like to do is, replace `Next Release` with the updated version, for example `1.3.0`. That way, when I release a new version, I don't need to remember that I need to change that title into the release I'm building.

Like I said, there are a few occasions where I ended up pushing `Next Release` to PyPi (here for example: https://pypi.org/project/django-auth-adfs/ )

**Comment by lgiordani** [comment]
I definitely see your need here. The way `serializer` works now, is that of searching and replacing with the same pattern, just using different versions to render the template. You want to decouple the `search` and `replace` like

``` python
GLOBALS = {
    'serializer': {
        'search': "next release",
        'replace': "{{year}}.{{month}}.{{day}}"
    }
}
```

I think this is a sensible use case, and actually something I was already considering as a possible extension of serializers. I'll work on it, I think I can have it in the next release that will be out in a week or so. Stay tuned, and thanks for the suggestion!

**Comment by lgiordani** [comment]
Hi! So the current develop contains a change that should implement what you requested. Can you please check the documentation at https://github.com/lgiordani/punch/blob/develop/docs/advanced.md#complex-serializers and tell me if it makes sense to you? Thanks! I will release develop as version 2.0.0 in a couple of days.

**Comment by jobec** [comment]
I looked at it, but I found it a bit a confusing.

Not the search/replace part, but the fact that `serializer` sometimes is a string, sometimes a list and in the example above a dictionary of dictionaries.
The string vs. list is understandable, but the confusing part is the key `semver` in the dictionary. I spent a couple of minutes figuring out if semver had a special meaning, of whether it could be anything. And also whether is was used somewhere else or not.

Honestly, I think a list of dictionaries of tuples makes it more clear.

e.g. from this:
```python
GLOBALS = {
    'serializer': {
        'semver': {
            'search': 'Next Release',
            'replace': '{{major}}.{{minor}}.{{patch}}'
        },
        'somethingelse': {
            'search': 'Unreleased',
            'replace': '{{major}}.{{minor}}.{{patch}}'
        }
    },
}
```
to this

```python
GLOBALS = {
    'serializer': [
        {
            'search': 'Next Release',
            'replace': '{{major}}.{{minor}}.{{patch}}'
        },
        {
            'search': 'Unreleased',
            'replace': '{{major}}.{{minor}}.{{patch}}'
        },
    ],
}
```
Or this
```python
GLOBALS = {
    'serializer': [
        ('Next Release', '{{major}}.{{minor}}.{{patch}}')
        ('Unreleased', '{{major}}.{{minor}}.{{patch}}')
    ],
}
```

**Comment by lgiordani** [comment]
You are right when you say that this is confusing, I double checked the documentation and it's not clear what you can do with serializers. As for the named serializers, they are there for a reason, namely to allow picking a specific one for the VCS, and they also allow to append serializers in specific files instead of overwriting all of them.

I will try to clarify it in the documentation, to see if it becomes more understandable. I'm not removing the named serializers however, they are part of the big change and that would mean to go back with other features as well.

Thanks! I'll post here a link to the new docs as soon as they are ready.

**Comment by lgiordani** [comment]
I tried to clarify the syntax of serializers here https://github.com/lgiordani/punch/blob/develop/docs/configuration.md#serializers. Thanks!

**Comment by jobec** [comment]
Only 2 remarks:
- There’s a typo in the 1.3.0 version in the example. It’s first written as 1.30.
- there’s a markup issue after the dictionary example

**Comment by lgiordani** [comment]
Great! Thanks, I fixed them now

**Comment by lgiordani** [comment]
Solved in version 2.0.0
